astropy/nddata/utils.py

```diff
@@ -456,3 +469,267 @@ def block_replicate(data, block_size, conserve_sum=True):
         data = data / float(np.prod(block_size))
 
     return data
+
+
+class Cutout(object):
+    def __init__(self, data, position, shape, wcs=None, mode='trim',
+                 fill_value=np.nan):
+        """
+        Create a cutout object from a 2D array.
+
+        The returned object will contain the 2D cutout array and the
+        updated WCS, if a `~astropy.wcs.WCS` object is input.
```

Comment:
I also added a `copy` option.  If `True` then `Cutout2D.data` will be a copy instead of a view.